src/drivers/dw/dma.c

```diff
@@ -336,6 +337,67 @@ static inline void dw_update_bits(struct dma *dma, uint32_t reg, uint32_t mask,
 	io_reg_update_bits(dma_base(dma) + reg, mask, value);
 }
 
+static inline void dw_dma_interrupt_mask(struct dma *dma, int channel)
+{
+	struct dma_pdata *p = dma_get_drvdata(dma);
+
+	if (!p->chan[channel].irq_enabled) {
+		tracev_event(TRACE_CLASS_DMA, "dw_dma_interrupt_mask(): "
+			     "dma %d channel %d not working in irq mode",
+			     dma->plat_data.id, channel);
+		return;
+	}
+
+	/* mask block, transfer and error interrupts for channel */
+	dw_write(dma, DW_MASK_TFR, INT_MASK(channel));
+	dw_write(dma, DW_MASK_BLOCK, INT_MASK(channel));
+	dw_write(dma, DW_MASK_ERR, INT_MASK(channel));
+}
+
+static inline void dw_dma_interrupt_unmask(struct dma *dma, int channel)
+{
+	struct dma_pdata *p = dma_get_drvdata(dma);
+
+	if (!p->chan[channel].irq_enabled) {
+		tracev_event(TRACE_CLASS_DMA, "dw_dma_interrupt_mask(): "
+			     "dma %d channel %d not working in irq mode",
+			     dma->plat_data.id, channel);
+		return;
+	}
+
+	/* unmask block, transfer and error interrupts
+	 * for channel
```

Comment:
I think that comment will easily fit within 80 characters, just put it in a single line